### DISCUSSION TITLE
Subject: Request for Help Building a Custom Version for SoLoGood F722

### DISCUSSION THREAD

**Discussion by TarekMlitan** [iNavFlight/inav#10493] [General]
Hello,

I am trying to build a custom version for my SoLoGood F722 board. I have attempted the bet2inav method to convert from Betaflight to iNav, but there are a few issues that I need help with.

The gyroscope and accelerometer are not functioning properly after the conversion.
Most of the UART ports do not seem to work as expected.
I would appreciate any advice or guidance on fixing these issues and ensuring that all components, like the gyroscope and UART, are correctly configured and functioning after the conversion.

Thank you in advance for your help!

Best regards,

**Comment by TarekMlitan** [comment]
FC Specifications:

Main control: STM32F722RET6

Gyroscope: ICM42688P

Barometer: Support (DPS310)

OSD: Support (AT7456E)

BEC: 10V/2A, 5V/2A

UART port: 6

Black box: onboard (16M flash memory)

Number of supported motors: M1-M8

Firmware name: BETAFLIGHT: SOLOGOODF722

**Comment by TarekMlitan** [comment]
betaflight resource
resource BEEPER 1 C13
resource MOTOR 1 C08
resource MOTOR 2 C09
resource MOTOR 3 A08
resource MOTOR 4 A09
resource MOTOR 5 B00
resource MOTOR 6 B01
resource MOTOR 7 A10
resource MOTOR 8 B04
resource LED_STRIP 1 B03
resource SERIAL_TX 1 B06
resource SERIAL_TX 2 A02
resource SERIAL_TX 3 B10
resource SERIAL_TX 4 A00
resource SERIAL_TX 5 C12
resource SERIAL_TX 6 C06
resource SERIAL_RX 1 B07
resource SERIAL_RX 2 A03
resource SERIAL_RX 3 B11
resource SERIAL_RX 4 A01
resource SERIAL_RX 5 D02
resource SERIAL_RX 6 C07
resource I2C_SCL 1 B08
resource I2C_SDA 1 B09
resource LED 1 C15
resource SPI_SCK 1 A05
resource SPI_SCK 2 B13
resource SPI_SCK 3 C10
resource SPI_SDI 1 A06
resource SPI_SDI 2 B14
resource SPI_SDI 3 C11
resource SPI_SDO 1 A07
resource SPI_SDO 2 B15
resource SPI_SDO 3 B05
resource ADC_BATT 1 C01
resource ADC_CURR 1 C03
resource PINIO 1 C02
resource PINIO 2 C00
resource FLASH_CS 1 A15
resource OSD_CS 1 B12
resource GYRO_EXTI 1 C04
resource GYRO_CS 1 A04

**Comment by TarekMlitan** [comment]
inav resource:
A00: FREE 
A01: FREE 
A02: FREE 
A03: FREE 
A04: SPI_PREINIT CS
A05: SPI1 SCK
A06: SPI1 MOSI
A07: SPI1 MISO
A08: MOTOR3 OUT
A09: MOTOR4 OUT
A10: FREE 
A11: USB IN
A12: USB OUT
A13: FREE 
A14: FREE 
A15: FLASH CS
B00: FREE 
B01: FREE 
B02: FREE 
B03: FREE 
B04: FREE 
B05: SPI3 MISO
B06: SERIAL1 UART TX
B07: SERIAL1 UART RX
B08: I2C1 SCL
B09: I2C1 SDA
B10: FREE 
B11: FREE 
B12: OSD CS
B13: SPI2 SCK
B14: SPI2 MOSI
B15: SPI2 MISO
C00: PINIO2 OUT
C01: ADC CH1
C02: PINIO1 OUT
C03: ADC CH2
C04: FREE 
C05: FREE 
C06: FREE 
C07: FREE 
C08: MOTOR1 OUT
C09: MOTOR2 OUT
C10: SPI3 SCK
C11: SPI3 MOSI
C12: FREE 
C13: BEEPER OUT
C14: FREE 
C15: LED1 OUT
D00: FREE 
D01: FREE 
D02: FREE 
D03: FREE 
D04: FREE 
D05: FREE 
D06: FREE 
D07: FREE 
D08: FREE 
D09: FREE 
D10: FREE 
D11: FREE 
D12: FREE 
D13: FREE 
D14: FREE 
D15: FREE 
E00: FREE 
E01: FREE 
E02: FREE 
E03: FREE 
E04: FREE 
E05: FREE 
E06: FREE 
E07: FREE 
E08: FREE 
E09: FREE 
E10: FREE 
E11: FREE 
E12: FREE 
E13: FREE 
E14: FREE 
E15: FREE 
F00: FREE 
F01: FREE 
F02: FREE 
F03: FREE 
F04: FREE 
F05: FREE 
F06: FREE 
F07: FREE 
F08: FREE 
F09: FREE 
F10: FREE 
F11: FREE 
F12: FREE 
F13: FREE 
F14: FREE 
F15: FREE

**Comment by sensei-hacker** [comment]
My response is on your Facebook thread and Discord.
I won't take the time to type it out a third time here since you haven't been responding.
